IOS/SwiftSeedProject/SwiftSeedProject/Application/Service/BaseService.swift

```diff
@@ -7,25 +7,34 @@
 //
 
 import Foundation
+import SwiftyJSON
 
 class BaseService<T: PersistenceObject>  {
     
-    private let persistence: Persistence
-    private let restClient: AnyObject? = nil
-    
-    //TODO: we should inclued the API access also
-    init(persistence: Persistence) {
+    private let persistence: Persistence!
+    internal let api: Api!
+
+    init(persistence: Persistence, api: Api) {
```

Comment:
Yep I don't like it either, I bet @mvazquez-ms will come with a good alternative